### PR TITLE
[ENH] Add different thresholding options to connectivity analysis

### LINKED ISSUE #40: The connectivity analysis should have different thresholding options
### Summary

The connectivity analysis provided in `ev.py` should have give the option to threshold the edge time-series matrix in a spatio-temporal  fashion. It should also allow for time-locked thresholding of the temporal and spatio-temporal thresholds.

### Next Steps
- Separate the thresholding step in `event_detection()` into an RSS-based and ETS-based thresholding.
- Refactor `event_detection()` to avoid duplicate lines of code between the RSS-based and ETS-based thresholding.
- Add an argument to the CLI to select the thresholding method.
- Add tests.

### PR DESCRIPTION
**Pull request by eurunuela**
Closes #40 .

Changes proposed in this pull request:

- Adds the option to threshold the edge time-series matrix with temporal and spatio-temporal criteria.

### CODE REVIEW

File: connPFM/tests/test_ev.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ def test_calculate_ets(ets_auc_original_file, AUC_file, atlas_file):
 
     AUC_img = masker.fit_transform(AUC_file)
     ets_auc_orig = np.loadtxt(ets_auc_original_file)
-    ets_auc_calc, u_vec, v_vec = ev.calculate_ets(AUC_img, AUC_img.shape[1])
```
Comment (vinferrer) [inline connPFM/tests/test_ev.py:19]: if you are moving the the `calculate _ets` function to `utils`, you should also move its testing to `test_utils.py` let's be coherent. Same goes for all the other moved functions

File: connPFM/tests/test_ev.py [hunk 2]
```diff
@@ -39,23 +39,15 @@ def test_threshold_ets_matrix():
     dum_mat[2, 1] = 3
     dum_mat2 = np.zeros((3, 3))
     dum_mat2[2, 1] = 3
-    th_dum = ev.threshold_ets_matrix(dum_mat, [2], 2)
+    th_dum = utils.threshold_ets_matrix(dum_mat, thr=2, selected_idxs=2)
     assert np.all(th_dum == dum_mat2)
 
 
 def test_event_detection(
```
Comment (vinferrer) [inline connPFM/tests/test_ev.py:46]: You should add the different rss options to the unittest testing so we don't lose coverage

File: connPFM/tests/test_ev.py (hunk 1 above)
Comment (eurunuela) [reply connPFM/tests/test_ev.py:19]: This is a different utils (it's inside the connectivity folder). We could create a `test_connectivity_utils.py` if that's better in your opinion.

File: connPFM/tests/test_ev.py (hunk 2 above)
Comment (eurunuela) [reply connPFM/tests/test_ev.py:46]: I'm planning on it. That's why this is still not ready for a review 😝

**Comment by eurunuela** [comment]
This is ready @vinferrer

**Review by vinferrer — COMMENTED** [review]
There is a few empty files that you commited that i believe are not necessary:
connPFM/connectivity/__init__.py
connPFM/debiasing/__init__.py
connPFM/deconvolution/__init__.py
connPFM/tests/__init__.py
Where you aware of them? what are they for? if they are being generated by the test, could you avoid it?

**Comment by vinferrer** [comment]
I am not finished

File: connPFM/connectivity/ev.py [hunk 3]
```diff
@@ -68,132 +40,128 @@ def event_detection(data_file, atlas, surrprefix="", sursufix="", nsur=100, segm
...
+    etspeaks = np.zeros(ets.shape)
 
-    # calculate rss
-    rss = np.sqrt(np.sum(np.square(ets), axis=1))
-    # initialize array for null rss
+    # Initialize array of indices of detected peaks
+    idxpeak = np.zeros(t)
+
+    # Initialize RSS with zeros
+    rss = np.zeros(t)
```
Comment (vinferrer) [inline connPFM/connectivity/ev.py:52]: why is this declared here and not inside the if condition in line 72

Comment (eurunuela) [reply connPFM/connectivity/ev.py:52]: Because the function returns RSS regardless of the option provided.

**Comment by eurunuela** [comment]
> There is a few empty files that you commited that i believe are not necessary:
> connPFM/connectivity/**init**.py
> connPFM/debiasing/**init**.py
> connPFM/deconvolution/**init**.py
> connPFM/tests/**init**.py
> Where you aware of them? what are they for? if they are being generated by the test, could you avoid it?

That was 100% intentional. I could not run the tests without that.

If you want to learn what they do, read this: https://stackoverflow.com/questions/448271/what-is-init-py-for

File: connPFM/connectivity/ev.py [hunk 4]
```diff
@@ -68,132 +40,128 @@ def event_detection(data_file, atlas, surrprefix="", sursufix="", nsur=100, segm
...
+                masker,
+                hist_range=(hist_min, hist_max),
+                numrand=nsur,
+            )
+        elif peak_detection == "ets_time":
+            # Initialize array for threshold
+            thr = np.zeros(t)
+
+            # Calculate ETS matrix of each surrogate
+            sur_ets = Parallel(n_jobs=-1, backend="multiprocessing")(
```
Comment (vinferrer) [inline connPFM/connectivity/ev.py:130]: suggestion, In the function `rss_surr` we already calculate ets of each surrogate to get the rss, can't you take it from that function. That way we are computationally more efficient and we avoid duplicating code

File: connPFM/tests/test_ev.py [hunk 5]
```diff
@@ -28,34 +28,110 @@ def test_rss_surr(AUC_file, atlas_file, surr_dir, rssr_auc_file):
...
-    rssr, _, _ = ev.rss_surr(AUC_img, u, v, join(surr_dir, "surrogate_AUC_"), "", masker, 0)
+    _, u, v = connectivity_utils.calculate_ets(AUC_img, AUC_img.shape[1])
+    rssr, _, _ = connectivity_utils.rss_surr(
+        AUC_img, u, v, join(surr_dir, "surrogate_AUC_"), "", masker, 0
+    )
     rssr_auc = np.loadtxt(rssr_auc_file)
     assert np.all(np.isclose(rssr, rssr_auc))
 
 
 def test_threshold_ets_matrix():
```
Comment (vinferrer) [inline connPFM/tests/test_ev.py:39]: Could you explain the logic behind this test?

**Review by vinferrer — CHANGES_REQUESTED** [review]
Nice work @eurunuela. However, could you make changes from the comments i left?

File: connPFM/connectivity/ev.py (hunk 3 above)
Comment (vinferrer) [reply connPFM/connectivity/ev.py:52]: makes sense

File: connPFM/tests/test_ev.py [hunk 6]
```diff
@@ -28,34 +28,110 @@ def test_rss_surr(AUC_file, atlas_file, surr_dir, rssr_auc_file):
...
+    thr = 2
     dum_mat = np.ones((3, 3))
     dum_mat[2, 1] = 3
     dum_mat2 = np.zeros((3, 3))
-    dum_mat2[2, 1] = 3
-    th_dum = ev.threshold_ets_matrix(dum_mat, [2], 2)
-    assert np.all(th_dum == dum_mat2)
+    dum_mat2[2, 1] = 3 - thr
```
Comment (vinferrer) [inline connPFM/tests/test_ev.py:44]: shouldn't this be dum_mat2
```suggestion
    dum_mat2[2, 1] = 1
```

File: connPFM/tests/test_ev.py (hunk 5 above)
Comment (eurunuela) [reply connPFM/tests/test_ev.py:39]: You wrote that test haha. I just moved the functions to `connectivity_utils`. But now that I see it, I think it makes more sense to join this test with the one above.

File: connPFM/connectivity/ev.py (hunk 4 above)
Comment (eurunuela) [reply connPFM/connectivity/ev.py:130]: Good point! Didn't see that. I will update `rss_surr` to return the ets matrix.

Comment (eurunuela) [reply connPFM/connectivity/ev.py]: `875c96b` should do this.

**Comment by eurunuela** [comment]
What do you think @vinferrer ? Are you happy with the changes?

**Review by vinferrer — APPROVED** [review]
Great job @eurunuela !!

**Pull request merged**